homeassistant/components/comfoconnect/climate.py

```diff
@@ -0,0 +1,135 @@
+    """Representation of the ComfoConnect bypass platform."""
+
+    def __init__(self, name, ccb: ComfoConnectBridge) -> None:
+        """Initialize the ComfoConnect bypass."""
+        self._name = name
+        self._ccb = ccb
+
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+        _LOGGER.debug("Registering for bypass state")
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_COMFOCONNECT_UPDATE_RECEIVED.format(SENSOR_BYPASS_STATE),
+                self._handle_update_bypass,
+            )
+        )
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_COMFOCONNECT_UPDATE_RECEIVED.format(SENSOR_TEMPERATURE_SUPPLY),
+                self._handle_update_temp,
+            )
+        )
+        await self.hass.async_add_executor_job(
+            self._ccb.comfoconnect.register_sensor, SENSOR_BYPASS_STATE
+        )
+        await self.hass.async_add_executor_job(
+            self._ccb.comfoconnect.register_sensor, SENSOR_TEMPERATURE_SUPPLY
+        )
+
+    def _handle_update_bypass(self, value):
+        """Handle update callback for bypass state."""
+        _LOGGER.debug(
+            "Handle update for climate - bypass (%d): %s", SENSOR_BYPASS_STATE, value
+        )
+        SENSOR_BYPASS_STATE_LOCAL = value
+
+        self._ccb.data[SENSOR_BYPASS_STATE] = SENSOR_BYPASS_STATE_LOCAL
+        self.schedule_update_ha_state()
+
+    def _handle_update_temp(self, value):
+        """Handle update callback for temperature."""
+        _LOGGER.debug(
+            "Handle update for climate - temp (%d): %s",
+            SENSOR_TEMPERATURE_SUPPLY,
+            value,
+        )
+        SENSOR_TEMPERATURE_SUPPLY_LOCAL = value
+
+        self._ccb.data[SENSOR_TEMPERATURE_SUPPLY] = SENSOR_TEMPERATURE_SUPPLY_LOCAL
+        self.schedule_update_ha_state()
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        _LOGGER.debug("set_hvac_mode: %s", hvac_mode)
+
+        if hvac_mode == HVAC_MODE_HEAT:
+            _LOGGER.debug("Send command: CMD_BYPASS_OFF")
+            self._ccb.comfoconnect.cmd_rmi_request(CMD_BYPASS_OFF)
+        elif hvac_mode == HVAC_MODE_COOL:
+            _LOGGER.debug("Send command: CMD_BYPASS_ON")
+            self._ccb.comfoconnect.cmd_rmi_request(CMD_BYPASS_ON)
+        else:
+            _LOGGER.error("Unsupported hvac_mode: %s", hvac_mode)
+
+        # Update current mode
+        self.schedule_update_ha_state()
+
+    @property
+    def hvac_modes(self):
+        """List of available bypass modes."""
+        return HVAC_MODES
+
+    @property
+    def hvac_mode(self):
+        """Return the current bypass mode. If the bypass state > 0 ComfoConenct started enabling bypass. 100% of bypass state = minimal heat recovery."""
+        bypass = SENSOR_BYPASS_STATE_LOCAL
```

Comment:
Check the instance attribute here.

Reply:
What do you mean? Instance of SENSOR_BYPASS_STATE_LOCAL? To make sure it's int?

Reply:
At the line below we have a check `if bypass:`.

`bypass` should be made an instance attribute instead of a reference to a constant since it's not a constant. We're modifying this variable.

Reply:
I see, will fix in next commit